.github/workflows/ci.yml

```
@@ -266,12 +266,22 @@ jobs:
         working-directory: test
 
   test-windows:
+    name: test-windows (${{ matrix.name }})
     needs: [lint, protos, verify-vendor, go-gen]
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ matrix.runner }}
     strategy:
       fail-fast: false
       matrix:
-        os: [windows-2019, windows-2022]
+        name:
+          [windows-2022, windows-2019]
+        runner:
+          - [self-hosted, 1ES.Pool=containerplat-github-runner-pool-east-us-2, 1ES.ImageOverride=github-mms-ws2022-containers-enabled]
+          - [self-hosted, 1ES.Pool=containerplat-github-runner-pool-east-us-2, 1ES.ImageOverride=github-mms-ws2019-containers-enabled]
+        exclude: # exclude windows-2022 job from running on ws2019 image and vice versa
+          - name: "windows-2022"
+            runner: [self-hosted, 1ES.Pool=containerplat-github-runner-pool-east-us-2, 1ES.ImageOverride=github-mms-ws2019-containers-enabled]
+          - name: "windows-2019"
+            runner: [self-hosted, 1ES.Pool=containerplat-github-runner-pool-east-us-2, 1ES.ImageOverride=github-mms-ws2022-containers-enabled]
```

probably cleaner to use `include` to [expand the definitions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#expanding-or-adding-matrix-configurations) so the `runner` settings only need to be repeated once:
```suggestion
        name:
          [windows-2022, windows-2019]
        include: 
          - name: "windows-2019"
            runner: [self-hosted, 1ES.Pool=containerplat-github-runner-pool-east-us-2, 1ES.ImageOverride=github-mms-ws2019-containers-enabled]
          - name: "windows-2022"
            runner: [self-hosted, 1ES.Pool=containerplat-github-runner-pool-east-us-2, 1ES.ImageOverride=github-mms-ws2022-containers-enabled]
```

---

Great suggestion, thanks!